src/data.rs

```diff
@@ -198,11 +192,11 @@ pub fn get_finality_sync_checkpoint(db: Arc<DB>) -> Result<Option<FinalitySyncCh
 
 	let result = db
 		.get_cf(&cf_handle, FINALITY_SYNC_CHECKPOINT_KEY.as_bytes())
-		.context("Couldn't get {key} from db")?;
+		.context("Couldn't get finality sync checkpoint from db")?;
 
 	result.map_or(Ok(None), |e| {
 		FinalitySyncCheckpoint::decode(&mut &e[..])
-			.map_err(|e| anyhow!("Error decoding: {e}!"))
+			.context("Failed to decoded finality sync checkpoint")
```

Comment:
Ditto.

Reply:
`.context` actually wraps error and still has old error there. The point is how you print it. No information is sacrificed.